remote_execution/oss/re_grpc/src/client.rs

```diff
@@ -478,13 +478,21 @@ impl FindMissingCache {
     }
 }
 
+#[derive(Clone)]
+enum OngoingUploadStatus {
+    Active(tokio::sync::watch::Receiver<Result<(), ()>>),
+    Done,
+    Error,
+}
+
 pub struct REClient {
     runtime_opts: RERuntimeOpts,
     grpc_clients: GRPCClients,
     capabilities: RECapabilities,
     instance_name: InstanceName,
     // buck2 calls find_missing for same blobs
     find_missing_cache: Mutex<FindMissingCache>,
+    prev_uploads: dashmap::DashMap<TDigest, OngoingUploadStatus>,
```

Comment:
Usual convention would be `use dashmap::DashMap` and then not qualify it onwards